types/opentok/index.d.ts

```diff
@@ -77,18 +79,70 @@ declare module 'opentok' {
       offset?: number;
       sessionId?: string;
     }
+
+    export interface BroadcastLayoutOptions {
+      type: 'bestFit' | 'pip' | 'verticalPresentation' | 'horizontalPresentation' | 'custom';
+      stylesheet?: string
+    }
+
+    export interface BroadcastOutputOptionsRtmp {
+      id: string;
+      serverUrl: string;
+      streamName: string;
+      status?: string;
+    }
+
+    export interface BroadcastOutputOptions {
+      hls?: {};
+      rtmp?: BroadcastOutputOptionsRtmp[];
+    }
+
+    export interface BroadcastOptions {
+      outputs: BroadcastOutputOptions;
+      maxDuration?: number;
+      resolution?: string;
+      layout: BroadcastLayoutOptions;
+    }
+
+    export interface BroadcastUrlsResponse {
+      hls?: string;
+      rtmp?: BroadcastOutputOptionsRtmp[];
+    }
+
+    export interface Broadcast {
+      id: string;
+      status: string;
+      broadcastUrls: BroadcastUrlsResponse;
+    }
+
+    export interface BroadcastStopResponse {
+      id: string;
+      sessionId: string;
+      projectId: number;
+      createdAt: number;
+      updatedAt: number;
+      resolution: string;
+    }
+
+    export interface SignalOptions {
+      type: string;
+      data: any;
+    }
   }
 
   class OpenTok {
     constructor(apiKey: string, apiSecret: string);
 
-    public createSession(options: OpenTok.SessionOptions, callback: (error: Error | null, session?: OpenTok.Session) => void): void;
+    public createSession(options: OpenTok.SessionOptions, callback: (error: Error, session: OpenTok.Session) => void): void;
```

Comment:
I'm not sure but I think the '| null' statements are for typescript transpiles that use the strict null checks and what not.

Reply:
To be safe I would leave the nulls alone.

Reply:
Ah.. yeah, I removed for testing, I should put it back.